packages/service-library/src/servicelib/async_utils.py

```diff
@@ -205,3 +212,40 @@ async def worker(in_q: Queue[QueueElement], out_q: Queue) -> None:
         return wrapper
 
     return decorator
+
+
+def with_delay(
+    delay: datetime.timedelta,
+) -> Callable[
+    [Callable[P, Coroutine[Any, Any, R]]], Callable[P, Coroutine[Any, Any, R]]
+]:
+    def decorator(
+        func: Callable[P, Coroutine[Any, Any, R]],
+    ) -> Callable[P, Coroutine[Any, Any, R]]:
+        @wraps(func)
+        async def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
+            await asyncio.sleep(delay.total_seconds())
+            return await func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+async def cancel_wait_task(
```

Comment:
MINOR: This reads
```python
await cancel_wait_task(t, max_delay=5)
```
alternative that might read a bit better?
```python
await force_cancel_task(t, timeout=5)
```
where `force` indicates that you do not expect an `uncancel`, the function awaits to cancel and has a `timeout` to perform the operation otherwise it will raise.

Reply:
actually I re-used the same reading as in asyncio internals. so I would guess they used common reading.
Also here I do not force anything, so I would not use force here, as I do standard cancel not force cancel.